### PR TITLE
Enable ES6 transpilation. Move HealthCare App to ES6 modules

### PR DESCRIPTION
**Pull request by awong-dev**
ES6 is the future. This lets us join it and be much cleaner in our Javascript.

### CODE REVIEW

File: assets/js/entry.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
-"use strict;"
+// Add all the ES6 library features that my be missing.
+require ("babel-polyfill");
```
Comment (akainic) [inline assets/js/entry.js]: extra space?

File: assets/js/entry.js [hunk 2]
```diff
@@ -12,10 +13,24 @@ if (!Modernizr.dataset) {
 
 // Bring in foundation and custom libraries.
 require('foundation/js/foundation/foundation');
-var WOW = require('wow.js/src/wow.coffee');
+let WOW = require('wow.js/src/wow.coffee');
```
Comment (akainic) [inline assets/js/entry.js]: For my own understanding, where are we deciding to use `let` vs. `var`? I was reading on ES6 and it looks like they are scoped differently, but it sounds like the recommendation is in general we should be using `let` going forward?

File: webpack_production.config.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// TODO: Remove this whole file in favor of a build envrionment variable like
+// used in definePlugin in webpack.config.js.
+
+var path = require('path');
+
+var config = require ('./webpack.config');
```
Comment (akainic) [inline webpack_production.config.js]: extra space?

**Comment by akainic** [comment]
LGTM! Just asked a question for my own understanding :)

File: assets/js/entry.js (hunk 2 above)
Comment (awong-dev) [inline assets/js/entry.js]: Yes.  `var` along with `this` and `==` are two of those JS "features" that were just broken. Like really stupid.

Basically, would you ever want:

```
function f () {
  console.log(a);
  var a = 0;
}
```
to actually run without error?  No. It's crazy that the language, *by specification of how `var` "hoisting" works* makes the above code literally means:

```
function f () {
  var a;
  console.log(a);
  a = 0;
}
```

w.t.f.

Let doesn't hoist, meaning it behave sanely like any other language.

**Pull request merged**